AK/QuickSort.h

```diff
@@ -115,10 +115,10 @@ void single_pivot_quick_sort(Iterator start, Iterator end, LessThan less_than)
         // Recur into the shorter part of the remaining data
         // to ensure a stack depth of at most log(n).
         if (i > size / 2) {
-            single_pivot_quick_sort(start + i, end, less_than);
+            single_pivot_quick_sort<Iterator, LessThan>(start + i, end, less_than);
             end = start + i - 1;
         } else {
-            single_pivot_quick_sort(start, start + i - 1, less_than);
+            single_pivot_quick_sort<Iterator, LessThan>(start, start + i - 1, less_than);
```

Comment:
This is explicitly passing a `Function<Out(In...) const&`. Previously, this function could not be used at all with `Function<Out(In...)>` because as noted below, it's not possible to use move-semantics in this case.